CONTRIBUTING.md

```diff
@@ -355,6 +355,8 @@ We recommend you use IntelliJ as your IDE. The code style template for the proje
 
 * When appropriate use Java 8 Stream API
 * Categorize errors when throwing an exception
+* **Tests**
+    * Avoid adding `Thread.sleep` in tests--these can fail due to environmental conditions, such as garbage collection or noisy neighbors in the CI environment.
```

Comment:
Is there a more generic rule here? E.g. do not write tests that rely on tasks completing within a fixed time period. If a test needs to interact with a service running in another thread, verify that the service is ready before starting the test.

Reply:
In practice, I believe this is the most succinct way to communicate this, since they inevitably involve a sleep.